File: Python/problem041.py
#! /usr/bin/env python3
'''
Problem 41 - Project Euler
http://projecteuler.net/index.php?section=problems&id=041
'''
from mathtools import isprime
from itertools import permutations

if __name__ == '__main__':
    digits = list(map(str, range(1,10)))
    while len(digits) > 2:
        for i in reversed(sorted(permutations(digits))):
            p = int(''.join(i))
            if isprime(p):
                print(p)
                exit(0)
        else:
            digits.pop()
    else:
        print('nothing')

File: Clojure/problem041.clj
#! /usr/bin/env clojure
; Problem 41 - Project Euler
; http://projecteuler.net/index.php?section=problems&id=41

(require '[mathtools :refer [primes]])
(def primes10m (primes 10000000))
(def ndigit
  (memoize (fn [n] (set (range 1 (inc n))))))

(defn pandigital? [x]
  (let [strx (str x)]
    (= (set (map #(- (int %) 48) strx))
       (ndigit (count strx)))))

(defn pppp [s]
  (do (println "---tsuuka---") s))

(->> primes10m
     (filter pandigital?)
     (apply max)
     (println))
